cpp/include/raft/spatial/knn/detail/ivf_flat_build.cuh

```diff
@@ -405,15 +406,15 @@ void save(const handle_t& handle, const std::string& filename, const index<T, Id
   write_scalar(of, index_.metric());
```

Comment:
It would be nice if all of these `write_*` could be moved to `raft/core/serialize.hpp` (and renamed to `serialize_*` and `deserialize_*`).